### PR TITLE
Genomic data counts study-view endpoint

### PR DESCRIPTION
**Pull request by h164654156465**
Implement backend changes in issue https://github.com/cBioPortal/cbioportal/issues/10257

- add /genomic-data-counts/fetch endpoint to study view controller
- support query CNA events by a list of gene specific
- return a list of GenomicDataCountItems

### CODE REVIEW

File: service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java [hunk 1]
```diff
@@ -186,6 +194,85 @@ public List<CopyNumberCountByGene> getCNAAlterationCountByGenes(List<String> stu
...
+
+                List<GeneMolecularData> geneMolecularDataList = molecularDataService.getMolecularDataInMultipleMolecularProfiles(mappedProfileIds, mappedSampleIds,
+                        stableIds.stream().map(Integer::parseInt).collect(Collectors.toList()), "SUMMARY");
+                    
+                List<GenomicDataCount> genomicDataCounts = geneMolecularDataList
+                    .stream()
+                    .collect(Collectors.groupingBy(GeneMolecularData::getValue))
+                    .entrySet()
+                    .stream()
+                    .filter(entry -> entry.getKey() != null && !entry.getKey().equals("NA") && entry.getValue().size() > 0)
```
Comment (kalletlak) [inline service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java:255]: are conditions specified here possible to happen?

File: web/src/main/java/org/cbioportal/web/parameter/CopyNumberCountFilter.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package org.cbioportal.web.parameter;
+
+public class CopyNumberCountFilter {
```
Comment (kalletlak) [inline web/src/main/java/org/cbioportal/web/parameter/CopyNumberCountFilter.java:3]: never used

File: service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java (hunk 1 above)
Comment (h164654156465) [reply service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java:255]: Yes. Sometime getKey(), aka alteration, will return "NA".

File: web/src/main/java/org/cbioportal/web/parameter/CopyNumberCountFilter.java (hunk 2 above)
Comment (h164654156465) [reply web/src/main/java/org/cbioportal/web/parameter/CopyNumberCountFilter.java:3]: removed

File: service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java [hunk 3]
```diff
@@ -186,6 +194,85 @@ public List<CopyNumberCountByGene> getCNAAlterationCountByGenes(List<String> stu
...
+            .collect(Collectors.toSet());
+
+        Map<String, Integer> geneSymbolIdMap = geneService
+            .fetchGenes(new ArrayList<>(hugoGeneSymbols), "HUGO_GENE_SYMBOL",
+                "SUMMARY")
+            .stream().collect(Collectors.toMap(Gene::getHugoGeneSymbol, Gene::getEntrezGeneId));
+        
+        return genomicDataFilters
+            .stream()
+            .flatMap(gdFilter -> {
```
[the remark this answers is not in the document]
Comment (h164654156465) [reply service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java:218]: There are warnings once I do regular map on gdFilter. When I change to flatMap and return stream object, those errors go away. Then I can build the project.

[the remark this answers is not in the document]
Comment (kalletlak) [reply service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java:218]: Makes sense

File: service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java (hunk 1 above)
Comment (kalletlak) [reply service/src/main/java/org/cbioportal/service/impl/StudyViewServiceImpl.java:255]: Yes, I `NA` should be it's own category

File: service/src/test/java/org/cbioportal/service/impl/StudyViewServiceImplTest.java [hunk 4]
```diff
@@ -255,6 +256,79 @@ public void getCNAAlterationCountByGenes() throws Exception {
...
+        
+        Mockito.when(molecularDataService.getMolecularDataInMultipleMolecularProfiles(
+            anyList(),
+            anyList(),
+            anyList(),
+            anyString()))
+            .thenReturn(geneMolecularData);
+        
+        List<GenomicDataCountItem> result = studyViewService.getCNAAlterationCountsByGeneSpecific(studyIds, sampleIds, genomicDataFilters);
+        Assert.assertEquals(2L, result.size());
```
Comment (dippindots) [inline service/src/test/java/org/cbioportal/service/impl/StudyViewServiceImplTest.java:321]: Just curious why use `2L` here?

**Review by dippindots — APPROVED** [review]
Welcome to the cBioPortal community @h164654156465! Your pull request looks good to me. All my comments are minor. @kalletlak It's good to address comments before merging. But please feel free to merge if you want me to include it in today's release (And address the comments later).

File: service/src/test/java/org/cbioportal/service/impl/StudyViewServiceImplTest.java (hunk 4 above)
Comment (kalletlak) [reply service/src/test/java/org/cbioportal/service/impl/StudyViewServiceImplTest.java:321]: count 2(Long)

Comment (dippindots) [reply service/src/test/java/org/cbioportal/service/impl/StudyViewServiceImplTest.java:321]: @kalletlak It's just `2` is a small number and the default result.size() type is int, so it looks strange to specify `2L` here.

**Pull request merged**